web/src/components/CategoryEntry.tsx

```diff
@@ -76,79 +75,64 @@ type PropsType = {
     if (!contentWithoutHtml) {
       return null
     }
-    const textToHighlight = this.contentMatcher.getMatchedContent(query, contentWithoutHtml, numWordsSurrounding)
-    if (textToHighlight == null) {
-      return null
-    }
-
-    return (
-      <ContentMatchItem
-        aria-label={textToHighlight}
-        searchWords={[query]}
-        autoEscape
-        sanitize={normalizeSearchString}
-        textToHighlight={textToHighlight}
-        highlightStyle={{ backgroundColor: theme.colors.backgroundColor, fontWeight: 'bold' }}
-      />
-    )
-  }
-
-  renderTitle(): ReactNode {
-    const { query, category, theme } = this.props
-    return (
-      <CategoryListItem>
-        <Highlighter
-          searchWords={query ? [query] : []}
-          aria-label={category.title}
-          autoEscape
-          sanitize={normalizeSearchString}
-          highlightStyle={{ backgroundColor: theme.colors.backgroundColor, fontWeight: 'bold' }}
-          textToHighlight={category.title}
-        />
-        <div style={{ margin: '0 5px', fontSize: '12px' }}>{this.getMatchedContent(NUM_WORDS_SURROUNDING_MATCH)}</div>
-      </CategoryListItem>
-    )
-  }
-
-  render(): ReactNode {
-    const { category } = this.props
-    return (
-      <Row>
-        <StyledLink to={category.path}>
-          <CategoryThumbnail alt='' src={category.thumbnail || iconPlaceholder} />
-          {this.renderTitle()}
-        </StyledLink>
-        {this.renderSubCategories()}
-      </Row>
-    )
-  }
+    const contentMatcher = new ContentMatcher()
+    return contentMatcher.getMatchedContent(query, contentWithoutHtml, NUM_WORDS_SURROUNDING_MATCH)
+  }, [contentWithoutHtml, query])
+
+  const SubCategories = subCategories.map(subCategory => (
+    <SubCategory key={subCategory.path} dir='auto'>
+      <StyledLink to={subCategory.path}>
+        <SubCategoryCaption aria-label={subCategory.title}>{subCategory.title}</SubCategoryCaption>
+      </StyledLink>
+    </SubCategory>
+  ))
+
+  const Title = (
+    <Highlighter
+      dir='auto'
+      searchWords={query ? [query] : []}
+      aria-label={category.title}
+      autoEscape
+      sanitize={normalizeSearchString}
+      highlightStyle={{ backgroundColor: theme.colors.backgroundColor, fontWeight: 'bold' }}
+      textToHighlight={category.title}
+    />
+  )
+
+  const Content = textToHighlight && (
+    <ContentMatchItem
+      aria-label={textToHighlight}
+      searchWords={[query]}
+      autoEscape
+      sanitize={normalizeSearchString}
+      textToHighlight={textToHighlight}
+      highlightStyle={{ backgroundColor: theme.colors.backgroundColor, fontWeight: 'bold' }}
+    />
+  )
+
+  return (
+    <Row>
+      <StyledLink to={category.path}>
+        <CategoryThumbnail alt='' src={category.thumbnail || iconPlaceholder} />
+        <CategoryListItem dir='auto'>
+          {Title}
+          <div style={{ margin: '0 5px', fontSize: '12px' }} dir='auto'>
```

Comment:
please use styled component  and optional we should define a fontSize in rem in `fonts.ts` with 0.75rem and replace the 12px values.

Reply:
not done in this PR so won't do